includes/admin/class-wc-rest-payments-orders-controller.php

```diff
@@ -156,6 +156,11 @@ public function create_customer( $request ) {
 				return new WP_Error( 'wcpay_missing_order', __( 'Order not found', 'woocommerce-payments' ), [ 'status' => 404 ] );
 			}
 
+			$allowed_order_statuses = [ 'pending', 'processing', 'on-hold' ];
```

Comment:
I would use a filter here as store owners can have custom order statuses (e.g. by using [this plugin](https://docs.woocommerce.com/document/woocommerce-order-status-manager/)).

```suggestion
			$allowed_order_statuses = apply_filters( 'wcpay_create_customer_allowed_order_statuses', [ 'pending', 'processing', 'on-hold' ] );
```

Reply:
@shendy-a8c : spinning off the idea from @htdat, we should probably break processing the order when it's in `completed, cancelled, refunded, failed` statuses instead. Otherwise, statuses from 3rd party integration will be breaking the charge here.

What do you think?

Reply:
> I would use a filter here as store owners can have custom order statuses

Thank you for the suggestion. Committed here 823d8df756456cd47f6a666095f78f0f16612c52.

Reply:
> we should probably break processing the order when it's in completed, cancelled, refunded, failed statuses instead. Otherwise, statuses from 3rd party extensions will be breaking the charge here.

@kalessil Do you mean because now we have a filter for the allowed order statuses, we have to make sure not to proceed creating customer for status `completed`, `cancelled`, `refunded`, `failed`? So, instead of an array of allowed order statuses, better to have _disallowed_ order statuses?

Reply:
> So, instead of an array of allowed order statuses, better to have disallowed order statuses?

@shendy-a8c, correct.

Reply:
Got it. Addressed 40d208b8ed97ff53fe18944d7a0756db33cc09e3.

Reply:
Awesome, thanks a lot ! I believe we can go ahead and merge this without further collecting feedback.